pkg/detector/detector.go

```diff
@@ -323,7 +324,15 @@ func (d *ResourceDetector) OnAdd(obj interface{}) {
 
 // OnUpdate handles object update event and push the object to queue.
 func (d *ResourceDetector) OnUpdate(oldObj, newObj interface{}) {
-	d.OnAdd(newObj)
+	switch newObj.(type) {
+	case *policyv1alpha1.PropagationPolicy:
+		// currently do nothing, since a policy's resource selector can not be updated.
+		return
+	case *policyv1alpha1.ClusterPropagationPolicy:
+		return
```

Comment:
If the update operation is not allowed, there is no chance to be here, right? 

In addition, it will block PP/CPP update on other fields(like `.spec.placement`).

Reply:
Oh yes I forgot the update is rejected by the webhook